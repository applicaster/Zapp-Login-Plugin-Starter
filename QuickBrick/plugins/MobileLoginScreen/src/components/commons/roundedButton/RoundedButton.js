import React, {Component} from 'react'
import propTypes from 'prop-types'
import {Text, View, TouchableHighlight, StyleSheet} from 'react-native'
import colors from '../../../style/color'

export default class RoundedButton extends Component {
  render() {
    const {text} = this.props
    return (
      <TouchableHighlight style={styles.wrapper}>
        <Text style={styles.buttonText}>{text}</Text>
      </TouchableHighlight>
    )
  }
}
RoundedButton.propTypes = {
  text: propTypes.string.isRequired,
}

const styles = StyleSheet.create({
  wrapper: {
    padding: 15,
    display: 'flex',
    borderRadius: 40,
    borderWidth: 1,
    borderColor: colors.black,
    backgroundColor: colors.grey,
  },
  buttonText: {
    fontSize: 16,
    width: '100%',
    textAlign: 'center',
  },
})
