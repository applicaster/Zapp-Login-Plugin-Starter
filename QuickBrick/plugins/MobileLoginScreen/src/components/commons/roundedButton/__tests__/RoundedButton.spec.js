import React from 'react'
import renderer from 'react-test-renderer'

import RoundedButton from '../RoundedButton'

describe('@RoundedButton', () => {
  describe('Render', () => {
    it('Render with default props', () => {
      const text = 'Test'
      const root = renderer.create(<RoundedButton text={text} />)
      expect(root.toJSON()).toMatchSnapshot()
    })
  })
})
