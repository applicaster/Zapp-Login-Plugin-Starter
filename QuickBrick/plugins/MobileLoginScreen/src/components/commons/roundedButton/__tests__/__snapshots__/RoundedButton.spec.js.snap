// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@RoundedButton Render Render with default props 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "borderColor": "#000000",
        "borderRadius": 40,
        "borderWidth": 1,
        "display": "flex",
        "padding": 15,
      },
      Object {
        "backgroundColor": "#CCCCCC",
      },
    ]
  }
>
  <Text
    style={
      Object {
        "fontSize": 16,
        "textAlign": "center",
        "width": "100%",
      }
    }
  >
    Test
  </Text>
</View>
`;
